src/Http/Http/src/Internal/ResponseCookies.cs

```diff
@@ -158,7 +158,14 @@ public void Delete(string key, CookieOptions options)
             bool pathHasValue = !string.IsNullOrEmpty(options.Path);
 
             Func<string, string, CookieOptions, bool> rejectPredicate;
-            if (domainHasValue)
+            if (domainHasValue && pathHasValue)
+            {
+                rejectPredicate = (value, encKeyPlusEquals, opts) =>
+                    value.StartsWith(encKeyPlusEquals, StringComparison.OrdinalIgnoreCase) &&
+                        value.IndexOf($"domain={opts.Domain}", StringComparison.OrdinalIgnoreCase) != -1 &&
```

Comment:
nit: in the future we can consider removing these string allocs, don't change this PR